### ISSUE TITLE
Fix request: use of ReentrantLock in NotProperlySynchronizingOnFieldWhileUsingGuardedBy

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#474]
Rule does not detect use of ReentrantLock. For example:

```java
import org.springframework.stereotype.Component;
import javax.annotation.concurrent.GuardedBy;
import java.util.concurrent.locks.Lock;
import java.util.concurrent.locks.ReentrantLock;
import java.util.regex.Pattern;

@Component
public class Foo {
    private final Lock lock = new ReentrantLock();

    @GuardedBy("lock")
    private Pattern pattern;
    
    public void doChangePattern(String text) {
        lock.lock();
        try {
            pattern = Pattern.compile(text);
        }
        finally {
            lock.unlock();
        }
    }
}
```

Possible way for solution: check for use of `lock.lock()` outside of surrounding try/finally, of typeIs `java.util.concurrent.locks.Lock`?

Should also added to other rules: `NotProperlySynchronizingOnFieldWhileUsingGuardedBy`, ...

Workaround is to use old fashioned Object lock with `synchronized` keywords. (But might not work well with virtual thread pinning.)

**Comment by jborgers** [comment]
Is is actually on NotProperlySynchronizingOnFieldWhileUsingGuardedBy, renaming issue.
